File: src/main/java/com/ngolajr/ispmedia/entities/Critica.java
package com.ngolajr.ispmedia.entities;

import com.fasterxml.jackson.annotation.JsonBackReference;
import jakarta.persistence.*;
import lombok.Data;

@Entity
@Data
public class Critica {
    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private long id;
    private int nota;
    @Column(length = 300)
    private String critica;
    //verificar
    @ManyToOne
    private Utilizador critico;
    @ManyToOne
    @JsonBackReference(value = "album-critica")
    private Album album;
}

File: ISPMEDIA-ANGULAR/src/app/entities/Critica.ts
import {User} from "./User";
import {Album} from "./Album";

export class Critica {
  id: number;
  nota: number;
  critica: string;
  critico:User;
  album:Album;

  constructor(id: number, nota: number, critica: string, critico:User, album:Album) {
    this.id = id;
    this.nota = nota;
    this.critica = critica;
    this.critico = critico;
    this.album = album;
  }
}
